libgloss/arc/hl/hl_argc.c

```diff
@@ -0,0 +1,30 @@
+/*********************************************************************
+Copyright  2020 Synopsys Inc.
+*********************************************************************/
+
+#include <stddef.h>
+
+#include "hl_toolchain.h"
+#include "hl_api.h"
+
+static int
+hl_argc (void)
+{
+  uint32_t ret;
+  volatile __uncached char *p;
+
+  p = hl_message (HL_SYSCALL_ARGC, ":i", (uint32_t *) &ret);
+
+  if (p == NULL)
+    ret = 0;
+
+  hl_delete ();
+
+  return ret;
+}
+
+int
+_argc (void)
+{
+  return hl_argc ();
```

Comment:
why do u have a local and a global function doing the same?

Reply:
Sometimes _fun functions do more, see _open, _gettimeofday, _clock. I want to be predicatable here, so hl_fun use HOSTLINK_SYSCALL directly when _fun is wrapper for hl_fun. Yes, in most cases we have just single call inside of _fun, but I don't see any problem here.

Reply:
Can be, if you compile with Os it may be that those functions will not be inlined, hence, you lose space and speed.